tools/src/main/java/org/apache/kafka/tools/VerifiableConsumer.java

```diff
@@ -529,6 +530,24 @@ private static ArgumentParser argParser() {
                 .metavar("TOPIC")
                 .help("Consumes messages from this topic.");
 
+        parser.addArgument("--group-protocol")
+                .action(store())
+                .required(false)
+                .type(String.class)
+                .setDefault(ConsumerConfig.DEFAULT_GROUP_PROTOCOL)
```

Comment:
I believe the idea is that `DEFAULT_GROUP_PROTOCOL` changes to "CONSUMER" at an AK major release boundary. That would have the side-effect that the behaviour of this code will change at that boundary, probably unexpectedly. I suggest that you set the default to one of the concrete values, rather than the config default, and then make a conscious change in this case to move it forwards to the new value. For example, you could use a default of "CLASSIC" for releases less than 4.0, and "CONSUMER" for releases 4.0 and later.

Reply:
Hi @AndrewJSchofield 👋 

> I believe the idea is that `DEFAULT_GROUP_PROTOCOL` changes to "CONSUMER" at an AK major release boundary. That would have the side-effect that the behaviour of this code will change at that boundary, probably unexpectedly.

As of 3.8 `DEFAULT_GROUP_PROTOCOL` is used to determine which implementation of the `KafkaConsumerDelegate` to instantiate and use. As you point out, in 4.0 `DEFAULT_GROUP_PROTOCOL` switches from `CLASSIC` to `CONSUMER`. Won't that change affect users in a **much** more substantial way?

> I suggest that you set the default to one of the concrete values, rather than the config default, and then make a conscious change in this case to move it forwards to the new value. For example, you could use a default of "CLASSIC" for releases less than 4.0, and "CONSUMER" for releases 4.0 and later.

Given that the core client defaults to `DEFAULT_GROUP_PROTOCOL`, wouldn't we want this tool to behave likewise?

Sorry, I'm very tired, so I'm probably missing your point entirely 😫

Reply:
AIUI, `VerifiableConsumer` is only used for the system tests, not end users. A given system test provides the values/arguments to `VerifiableConsumer` as part of its test. Many tests that use the consumer have been retrofitted to test both consumer types. However, if a test doesn't explicitly provide a value for the group protocol, it is delegating to the `ConsumerConfig` to use whatever defaults it chooses.

Reply:
Apologies @kirktrue. I now realise that this PR is merely a backport for 3.7 and my comments really were addressed at the behaviour of the test in the future as the default changes. Happy to approve.

Reply:
Thanks for the review @AndrewJSchofield 😄 

Also, please file a bug against 4.0.0 if you feel the current behavior with defaults to be incorrect. Thanks!